### PR TITLE
fix: rename Entity::setAttributes() to avoid name collision

### LINKED ISSUE #5762: Bug: Entity can't handle column named `attributes`
### CodeIgniter4 Version

4.1.9

### What happened?

When using ``attributes``, as the property/attribute name of the Entity class, the `Entity::setAttributes()` method is called as a setter.
Names collision.

### Steps to Reproduce

```php
$e = new CodeIgniter\Entity\Entity();
$e->a = 1;
$e->attributes = [1, 2, 3];  // called the setAttributes method as a setter.
dd($e->toRawArray());

//$e->toRawArray() array (3)
//   0 => integer 1
//   1 => integer 2
//   2 => integer 3
```

### Expected Output

```
//$e->toRawArray() array (2)
//    a => integer 1
//    attributes => array (3)
//        0 => integer 1
//        1 => integer 2
//        2 => integer 3
```

### Anything else?

In Laravel prior to version 9, the custom setter and getter had the following format:
set/get + AttributeName + Attribute.
```php
public function getFullNameAttribute()
{
    return "{$this->first_name} {$this->last_name}";
}
```

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes #5762

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Entity/Entity.php [hunk 1]
```diff
@@ -274,7 +274,7 @@ public function hasChanged(?string $key = null): bool
      *
      * @return $this
      */
-    public function setAttributes(array $data)
+    public function injectRawData(array $data)
```
Comment (mostafakhudair) [inline system/Entity/Entity.php:277]: removing public function is BC, isn't it?

Comment (kenjis) [reply system/Entity/Entity.php:277]: Yes. I added `beraking change` label.

**Comment by MGatner** [comment]
@kenjis Is this still relevant after #5509?

**Comment by kenjis** [comment]
At least, this PR won't be merged into 4.x.
But I want to *fix* this bad design in the future.

**Comment by najdanovicivan** [comment]
I like the injectRawData naming. It can be merged into 4.x if setAttributes is still kept as deprecated so that we can upgrade before the next major version.

**Comment by kenjis** [comment]
I sent PR #7208

**Pull request closed without merging**